File: api/src/main/java/com/calendar/dao/ScheduleDao.java
package com.calendar.dao;

import com.calendar.entity.Schedule;
import org.seasar.doma.*;
import org.seasar.doma.boot.ConfigAutowireable;
import org.seasar.doma.jdbc.Result;

import java.util.List;

@ConfigAutowireable
@Dao
public interface ScheduleDao {

    @Insert
    Result<Schedule> insert(Schedule entity);

    @Select
    List<Schedule> findAll();

    @Select
    Schedule findById(int id);

    @Update
    Result<Schedule> update(Schedule entity);

    @Delete(sqlFile = true)
    int deleteById(int id);
}

File: src/main/kotlin/com/calendar/calendar/repository/ScheduleDao.kt
package com.calendar.calendar.repository

import com.calendar.calendar.entity.Schedule
import org.seasar.doma.Dao
import org.seasar.doma.Select
import org.seasar.doma.boot.ConfigAutowireable

@ConfigAutowireable
@Dao
interface ScheduleDao {

    @Select
    fun selectAll(): List<Schedule>
}